core/zoe/utils.js

```diff
@@ -0,0 +1,32 @@
+// used to reduce boolean arrays
+const bothTrue = (prev, curr) => prev && curr;
+const anyTrue = (prev, curr) => prev || curr;
+
+// https://stackoverflow.com/questions/17428587/transposing-a-2d-array-in-javascript/41772644#41772644
+const transpose = matrix =>
+  matrix.reduce(
+    (acc, row) => row.map((_, i) => [...(acc[i] || []), row[i]]),
+    [],
+  );
+
+const ruleEqual = (leftRule, rightRule) => leftRule.rule === rightRule.rule;
```

Comment:
I would be very surprised if `===` is what you want here, rather than `sameStructure`

Reply:
No, this is actually correct. From the names, I thought you were comparing rules. But you're actually comparing strings representing the kind of rule. `===` is good. Resolving.